src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,87 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidUserDataException;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+    private User defaultUser;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        defaultUser = createDefaultUser();
+    }
+
+    private User createDefaultUser() {
+        User user = new User();
+        user.setLogin("yaroslav");
+        user.setPassword("password123");
+        user.setAge(22);
```

Comment:
```suggestion
        user.setAge(18);
```
you don't check this edge case